lib/charms/mongodb/v0/mongodb_backups.py

```diff
@@ -239,6 +243,28 @@ def _on_create_backup_action(self, event) -> None:
             event.fail(f"Failed to backup MongoDB with error: {str(e)}")
             return
 
+    def _on_list_backups_action(self, event) -> None:
+        if self.model.get_relation(S3_RELATION) is None:
+            event.fail("Relation with s3-integrator charm missing, cannot list backups.")
+            return
+
+        # cannot list backup if pbm is not resynced and ready
+        try:
+            self._verify_resync()
+        except ResyncError:
+            event.defer()
+            logger.debug(
+                "Sync-ing configurations needs more time, must wait before listing backups."
+            )
+            return
```

Comment:
Shouldn't you also stop the flow of execution if `_verify_resync` sets a `BlockedStatus` due to incompatible S3 settings or invalid credentials?

Reply:
I think we should exit with an error if invalid credentials

Reply:
gotcha

Reply:
agreed